worker/runtime/process.go

```diff
@@ -58,6 +61,7 @@ func (p *Process) Wait() (int, error) {
 		return 0, fmt.Errorf("delete process: %w", err)
 	}
 
+	p.c.SetProperty(ProcessExitStatusKey, fmt.Sprintf("%d", status.ExitCode()))
```

Comment:
Yes, this is exactly what I was thinking of using! This looks good, continue down this path.